CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatChannel.lua

```diff
@@ -0,0 +1,163 @@
+--//////////////////////////////////////
+local modulesFolder = script.Parent
+local moduleTransparencyTweener = require(modulesFolder:WaitForChild("TransparencyTweener"))
+local moduleMessageLabelCreator = require(modulesFolder:WaitForChild("MessageLabelCreator"))
+local ClassMaker = require(modulesFolder:WaitForChild("ClassMaker"))
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+local methods = {}
+
+local function CreateGuiObject()
+	local BaseFrame = Instance.new("Frame")
+	BaseFrame.Size = UDim2.new(1, 0, 1, 0)
+	BaseFrame.BackgroundTransparency = 1
+
+	local Scroller = Instance.new("ScrollingFrame", BaseFrame)
+	Scroller.Name = "Scroller"
+	Scroller.BackgroundTransparency = 1
+	Scroller.BorderSizePixel = 0
+	Scroller.Position = UDim2.new(0, 0, 0, 3)
+	Scroller.Size = UDim2.new(1, -4, 1, -6)
+	Scroller.CanvasSize = UDim2.new(0, 0, 0, 0)
+	Scroller.ScrollBarThickness = module.ScrollBarThickness
+	Scroller.Active = false
+
+	return BaseFrame, Scroller
+end
+
+function methods:Destroy()
+	self.GuiObject:Destroy()
+end
+
+function methods:SetActive(active)
+	self.GuiObject.Visible = active
+end
+
+function methods:AddMessageLabelToLog(messageObject)
+	self.TextTweener:RegisterTweenObjectProperty(messageObject.Tweener, "Transparency")
+
+	table.insert(self.MessageObjectLog, messageObject)
+	self:PositionMessageLabelInWindow(messageObject)
+
+	if (#self.MessageObjectLog > 50) then
+		self:RemoveLastMessageLabelFromLog()
+	end
+end
+
+function methods:RemoveLastMessageLabelFromLog()
+	local lastMessage = self.MessageObjectLog[1]
+	local posOffset = UDim2.new(0, 0, 0, lastMessage.BaseFrame.AbsoluteSize.Y)
+
+	lastMessage:Destroy()
+	table.remove(self.MessageObjectLog, 1)
+
+	for i, messageObject in pairs(self.MessageObjectLog) do
+		messageObject.BaseFrame.Position = messageObject.BaseFrame.Position - posOffset
+	end
+
+	self.Scroller.CanvasSize = self.Scroller.CanvasSize - posOffset
+end
+
+function methods:PositionMessageLabelInWindow(messageObject)
+	local baseFrame = messageObject.BaseFrame
+	local baseMessage = messageObject.BaseMessage
+
+	baseFrame.Parent = self.Scroller
+	baseFrame.Position = UDim2.new(0, 0, 0, self.Scroller.CanvasSize.Y.Offset)
+
+	local textBoundsSize = moduleMessageLabelCreator:GetStringTextBounds(baseMessage.Text, baseMessage.Font, baseMessage.FontSize, UDim2.new(0, baseFrame.AbsoluteSize.X, 0, 1000))
+	baseFrame.Size = UDim2.new(1, 0, 0, textBoundsSize.Y)
+
+	local scrollBarBottomPosition = (self.Scroller.CanvasSize.Y.Offset - self.Scroller.AbsoluteSize.Y)
+	local reposition = (self.Scroller.CanvasPosition.Y >= scrollBarBottomPosition)
+
+	local add = UDim2.new(0, 0, 0, baseFrame.Size.Y.Offset)
+	self.Scroller.CanvasSize = self.Scroller.CanvasSize + add
+
+	if (reposition) then
+		self.Scroller.CanvasPosition = Vector2.new(0, math.max(0, self.Scroller.CanvasSize.Y.Offset - self.Scroller.AbsoluteSize.Y))
+	else
+		local displayNewMessagesIfNotFullyScrolledDown = false
```

Comment:
this is always false, lets just pass false here. If needed, add a comment for clarity